src/ProgramTest.elm

```diff
@@ -1048,17 +1075,12 @@ clickLink linkText href programTest =
             -> ProgramTest model msg effect
             -> ProgramTest model msg effect
         tryClicking { otherwise } =
-            andThen <|
-                \program state ->
-                    let
-                        link =
-                            Program.renderView program state.currentModel
-                                |> Query.find findLinkTag
-                    in
-                    if respondsTo normalClick link then
-                        -- there is a click handler
-                        -- first make sure the handler properly respects "Open in new tab", etc
-                        if respondsTo ctrlClick link || respondsTo metaClick link then
+            if respondsTo normalClick then
```

Comment:
I'm having a little bit of trouble convincing myself this refactoring is correct now that `repondsTo` doesn't take the found link element as an input.  I guess it's that `repondsTo` is re-running the original query, just modified to do the simulate as the last step of each branch?

Apart from being a bit confusing to read, it makes me a little nervous about the performance if we're re-doing that work, since ComplexQuery can be quite slow (especially when a query fails).

Here's one possible thought: maybe we could extract a new function from `simulateComplexQuery`/`assertComplexQuery`:

```elm
runComplexQuery :
    String
    -> (ComplexQuery (Query.Single msg) -> ComplexQuery a)
    -> (Program model msg effect (SimulatedSub msg) -> TestState model msg effect -> a ->  Result Failure (TestState model msg effect))
    -> ProgramTest model msg effect
    -> ProgramTest model msg effect

simulateComplexQuery functionName complexQuery =
    runComplexQuery functionName complexQuery <| \program state msg -> TestState.update msg program state

assertComplexQuery functionName complexQuery =
    runComplexQuery functionName complexQuery <| _ state _ -> state
```

And then here we could use `runComplexQuery` with the `findLinkQuery` to get the single, and a function that will take the single and use it to "try clicking", and then we wouldn't have to re-find the single in each call of `respondsTo`?

Reply:
Good point! Not sure what I was thinking implementing it that way. I think it was running the query five times! I actually later came up with something along the lines of your suggestion independently. I've now implemented `runComplexQuery` (with a slight change of argument order) and pushed.

`assertComplexQuery` is not being used now. Not sure if it's worth keeping it around.

Reply:
Yeah, there have been many times on this project where I've implemented something, and then when I show it to someone they ask why I did it so complicated 😄 

It's fine to leave `assertComplexQuery` for now, especially since I haven't finished reviewing if there are more functions that should be ported over to using ComplexQuery internally.